packages/http/libraries/core/src/http/decorators/SetHeader.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  getReflectMetadata,
+  setReflectMetadata,
+} from '@inversifyjs/reflect-metadata-utils';
+
+import { controllerMethodHeaderMetadataReflectKey } from '../../reflectMetadata/data/controllerMethodHeaderMetadataReflectKey';
+import { ControllerFunction } from '../models/ControllerFunction';
+
+export function setHeader(key: string, value: string): MethodDecorator {
+  return (
+    _target: object,
+    _key: string | symbol,
+    descriptor: PropertyDescriptor,
+  ): void => {
+    const headerMetadata: Map<string, string> =
+      getReflectMetadata(
+        descriptor.value as ControllerFunction,
+        controllerMethodHeaderMetadataReflectKey,
+      ) ?? new Map<string, string>();
+
+    headerMetadata.set(key, value);
```

Comment:
headers are not case sensitive, consider https://github.com/fastify/help/issues/71
```suggestion
    headerMetadata.set(key.toLowerCase(), value);
```

Reply:
Fixed